src/mon/OSDMonitor.cc

```diff
@@ -5797,6 +5797,21 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
+int OSDMonitor::check_pg_num(int pg_num, int size, ostream *ss)
+{
+  int64_t max_pgs_per_osd = g_conf->mon_pg_warn_max_per_osd;
+  int64_t max_pgs = max_pgs_per_osd * osdmap.get_num_osds();
+  int64_t max_pg_num = MAX(1, max_pgs / size);
+  if (pg_num > max_pg_num) {
```

Comment:
shouldn't we check total PGs, rather than just those in this pool?

Reply:
might get confusing with multiple crush roots, I guess we could account for those as well...